src/dotty/tools/dotc/transform/Pickler.scala

```diff
@@ -68,7 +68,7 @@ class Pickler extends Phase {
   private def pickleSourcefile(pickler: TastyPickler, source: SourceFile): Unit = {
     val buf = new TastyBuffer(10)
     pickler.newSection("Sourcefile", buf)
-    buf.writeNat(pickler.nameBuffer.nameIndex(source.file.path).index)
+    buf.writeNat(pickler.nameBuffer.nameIndex(source.file.canonicalPath).index)
```

Comment:
ExpandPrivate contains an assertion that compares two source files for equality. It only works when we canonicalize the paths. We could weaken the assertion and demand e.g. that the paths have a common non-trivial suffix. I'll make that change.